subprojects/platform-jvm/src/main/java/org/gradle/api/java/archives/internal/DefaultManifestMergeDetails.java

```diff
@@ -55,6 +55,13 @@ public String getMergeValue() {
 
     @Override
     public String getValue() {
+        //todo: issue warning
+        //todo: remove usages from test
+        return value.toString();
```

Comment:
I think this should work as follows:
- if the value is Provider then print warning about early eval and return value.get()
- otherwise, return value.toString()